org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/XtextGeneratorLanguage.java

```diff
@@ -0,0 +1,380 @@
+import org.eclipse.emf.common.util.Diagnostic;
+import org.eclipse.emf.common.util.DiagnosticChain;
+import org.eclipse.emf.common.util.TreeIterator;
+import org.eclipse.emf.common.util.URI;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.emf.ecore.EReference;
+import org.eclipse.emf.ecore.EValidator;
+import org.eclipse.emf.ecore.resource.Resource;
+import org.eclipse.emf.ecore.resource.ResourceSet;
+import org.eclipse.emf.ecore.util.EcoreUtil;
+import org.eclipse.xtext.AbstractMetamodelDeclaration;
+import org.eclipse.xtext.Grammar;
+import org.eclipse.xtext.GrammarUtil;
+import org.eclipse.xtext.ReferencedMetamodel;
+import org.eclipse.xtext.XtextPackage;
+import org.eclipse.xtext.nodemodel.INode;
+import org.eclipse.xtext.nodemodel.util.NodeModelUtils;
+import org.eclipse.xtext.resource.IResourceDescription;
+import org.eclipse.xtext.resource.IResourceServiceProvider;
+import org.eclipse.xtext.resource.XtextResource;
+import org.eclipse.xtext.resource.impl.ResourceDescriptionsData;
+import org.eclipse.xtext.xbase.lib.CollectionLiterals;
+import org.eclipse.xtext.xtext.RuleNames;
+import org.eclipse.xtext.xtext.generator.model.GuiceModuleAccess;
+import org.eclipse.xtext.xtext.generator.model.StandaloneSetupAccess;
+import org.eclipse.xtext.xtext.generator.model.project.IXtextProjectConfig;
+
+import com.google.common.base.Joiner;
+import com.google.common.collect.Lists;
+import com.google.inject.Binder;
+import com.google.inject.Inject;
+import com.google.inject.Injector;
+import com.google.inject.Provider;
+
+/**
+ * Configuration for an Xtext language. Use this class if you need to specify
+ * exactly which generator fragments you want to include, or alternatively
+ * {@link StandardLanguage} to work with the standard fragments.
+ * 
+ * <p>
+ * The most important property is {@code name}, which should be set to the
+ * language name exactly as specified in the grammar file. If the default
+ * grammar resolution based on the name does not work in your particular
+ * scenario, set the {@code grammarUri} property instead. Furthermore, use
+ * {@code fileExtensions} to specify the file extensions for your language.
+ * </p>
+ * 
+ * <p>
+ * Generator fragments can be added with the {@code fragment} property. See
+ * {@link StandardLanguage} for a list of useful fragments.
+ * </p>
+ * 
+ * @noextend This class should not be extended by clients.
+ */
+public class XtextGeneratorLanguage extends CompositeGeneratorFragment2 implements IXtextGeneratorLanguage {
+	private static final Logger LOG = Logger.getLogger(XtextGeneratorLanguage.class);
+
+	private String grammarUri;
+
+	private String name;
+
+	private Boolean generateXtendStubs;
+
+	private Grammar grammar;
+
+	private RuleNames ruleNames;
+
+	private List<String> fileExtensions;
+
+	private List<String> referencedResources = CollectionLiterals.<String>newArrayList();
+
+	private ResourceSet resourceSet;
+
+	private com.google.inject.Module guiceModule = (Binder it) -> {
```

Comment:
Could be an import.

Reply:
we have it this way for java 11+ compatibility

Reply:
Hmm java.lang.Module is not used in this class. I understand that we do this in generated code though. Anyways: Cosmetics :)